src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -127,65 +141,150 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
   };
 
+  const handleDownArrow = () => {
+    if (cursor.current + 1 < visibleNodes.length) {
+      cursor.current += 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  const handleUpArrow = () => {
+    if (cursor.current >= 1) {
+      cursor.current -= 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds parent of the current node, used for left arrow functionality
+   */
+  const findParentNode = () => {
+    if (!currentNodeId.current) {
+      return;
+    }
+    const parentId = visibleNodes.find((el) => el.id === currentNodeId.current).parent;
+    if (parentId !== '') {
+      cursor.current = visibleNodes.findIndex((el) => el.id === parentId);
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds the first node starting with the given character.
+   * Starts at the cursors current position, and wraps around to the beginning of the menu if a match isn't found.
+   * @param {*} char The character to search by
+   */
+  const findNodeMatching = (char) => {
+    let sortedNodes = visibleNodes.slice(cursor.current + 1, visibleNodes.length);
+    sortedNodes = sortedNodes.concat(visibleNodes.slice(0, cursor.current));
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === match.id);
+      currentNodeId.current = match.id;
+      focusCurrentNode();
+    }
+  };
+
   const handleKeyDown = (event, item) => {
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      event.preventDefault();
-      handleOnClick(event, item);
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        event.preventDefault();
+        handleOnClick(event, item);
+        break;
+      case KeyCode.KEY_DOWN:
+        event.preventDefault();
+        handleDownArrow();
+        break;
+      case KeyCode.KEY_UP:
+        event.preventDefault();
+        handleUpArrow();
+        break;
+      case KeyCode.KEY_RIGHT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).ariaExpanded === 'true') {
+            handleDownArrow();
+          } else if (document.getElementById(currentNodeId.current).ariaHasPopup && (!document.getElementById(currentNodeId.current).ariaExpanded || document.getElementById(currentNodeId.current).ariaExpanded === 'false')) {
+            handleOnClick(event, item);
+          }
+        }
+        break;
+      case KeyCode.KEY_LEFT:
+        event.preventDefault();
+        if (currentNodeId.current) {
+          if (document.getElementById(currentNodeId.current).ariaExpanded === 'true') {
```

Comment:
I believe this syntax will only work on chrome and safari. Firefox doesn't appear to function with it, and I would question IE as well.
It would likely be more reliable to switch to:
document.getElementById(currentNodeId.current).getAttribute('aria-expanded') === 'true'

Reply:
I haven't checked against IE, but I've implemented using `getAttribute` and it seems to be working on firefox now.